File: python/leetcode026.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ls = len(nums)
        if ls == 0:
            return 0
        k = 0
        j = 1
        for i in range(1, ls):
            if nums[i] == nums[k]:
                continue
            else:
                nums[j] = nums[i]
                k = i
                j = j + 1
        return j



if __name__ == "__main__":
    s = Solution()
    print(s.removeDuplicates([1,1,2]))

File: leetcodec++/leetcode026.cpp
//
// Created by 黄敏辉 on 2019-04-30.
//

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count = 0;
        for (int i = 0; i < nums.size(); i++){
            if (nums[i] == nums[i-1])
                count ++;
            else
                nums[i - count] = nums[i];
        }
        return nums.size() - count;
    }
};